File: TOMIMSV4/Entity/Interface/Pedido_Compra/Pedido_Compra_Det_Lotes/clsBeI_nav_ped_compra_det_lote.vb
Public Class clsBeI_nav_ped_compra_det_lote
    Implements ICloneable
    Public Property NoEnc() As String = ""
    Public Property source_ID() As String = ""
    Public Property Source_Prod_Order_Line() As Integer = 0
    Public Property Item_No() As String = ""
    Public Property Lot_No() As String = ""
    Public Property Expiration_Date() As Date = Date.Now
    Public Property Entry_No() As String = ""
    Public Property Source_Type() As Integer = 0
    Public Property Quantity_Base() As Double = 0.0
    Public Property Variant_Code() As String = ""

    Sub New()
    End Sub

    Sub New(ByRef NoEnc As String, ByVal source_IDField As String, ByVal Source_Prod_Order_Line As Integer, ByVal Item_No As Integer, ByVal Lot_No As String, ByVal Expiration_Date As Date, ByVal Entry_No As String, ByVal Source_Type As Integer, ByVal Quantity_Base As Double, ByVal Variant_Code As String)
        Me.NoEnc = NoEnc
        Me.source_ID = Source_IDField
        Me.Source_Prod_Order_Line = Source_Prod_Order_Line
        Me.Item_No = Item_No
        Me.Lot_No = Lot_No
        Me.Expiration_Date = Expiration_Date
        Me.Entry_No = Entry_No
        Me.Source_Type = Source_Type
        Me.Quantity_Base = Quantity_Base
        Me.Variant_Code = Variant_Code
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/I_nav_Ped_Compra/clsBeI_nav_ped_compra_det_lote.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeI_nav_ped_compra_det_lote : ICloneable
    {
        [Column("NoEnc")]
        [DisplayName("NoEnc")]
        public string NoEnc { get; set; } = "";

        [Column("source_ID")]
        [DisplayName("source_ID")]
        public string Source_ID { get; set; } = "";

        [Column("Source_Prod_Order_Line")]
        [DisplayName("Source_Prod_Order_Line")]
        public int Source_Prod_Order_Line { get; set; } = 0;

        [Column("Item_No")]
        [DisplayName("Item_No")]
        public string Item_No { get; set; } = "";

        [Column("Lot_No")]
        [DisplayName("Lot_No")]
        public string Lot_No { get; set; } = "";

        [Column("Expiration_Date")]
        [DisplayName("Expiration_Date")]
        public DateTime Expiration_Date { get; set; } = new DateTime(1900,1,1);

        [Column("Entry_No")]
        [DisplayName("Entry_No")]
        public string Entry_No { get; set; } = "";

        [Column("Source_Type")]
        [DisplayName("Source_Type")]
        public int Source_Type { get; set; } = 0;

        [Column("Quantity_Base")]
        [DisplayName("Quantity_Base")]
        public double Quantity_Base { get; set; } = 0;

        [Column("Variant_Code")]
        [DisplayName("Variant_Code")]
        public string Variant_Code { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public Double Pallet_Weight { get; set; } = 0;
        public string Pallet_License_No { get; set; } = "";

        public clsBeI_nav_ped_compra_det_lote() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
